### PR TITLE
Os/emailvalidation

### PR DESCRIPTION
**Pull request by taygetea**
Added a function to do email validation in questionbot, using ask_freeform_question as a base. Tried to distill the format into something repeatable and structured. Also added a test. There's potential to abstract this out, to make a function like ask_specific_question that takes question_text and a validator function and the re-prompt if invalid, and generates a new question type.

### CODE REVIEW

File: tests/testbots/test_questions_bot.py [hunk 1]
```diff
@@ -141,6 +141,18 @@ async def do_test_address_question_with_confirmation(
...
+    async def do_test_email_question(
+        self, message: Message
+    ) -> Response:
+
+        email = await self.ask_email_question(message.source)
+
+        if email:
+            return email
+        return "oops, sorry"
+
```
Comment (technillogue) [inline tests/testbots/test_questions_bot.py:153]: can you add the equivalent of this as a unit test?

File: forest/core.py [hunk 2]
```diff
@@ -1511,6 +1512,52 @@ async def ask_multiple_choice_question(  # pylint: disable=too-many-arguments
...
+                question_text = (
+                    "Please reply with a valid email address \n \n" + question_text
+                )
+
+            return await self.ask_email_question(
+                recipient,
+                question_text,
+                require_first_device,
+            )
+        # ----RETURN----
```
Comment (technillogue) [inline forest/core.py:1558]: maybe an excessive comment

File: forest/core.py [hunk 3]
```diff
@@ -1511,6 +1512,52 @@ async def ask_multiple_choice_question(  # pylint: disable=too-many-arguments
...
+        recipient: str,
+        question_text: Optional[str] = "Please enter your email address",
+        require_first_device: bool = False,
+    ) -> Optional[str]:
+        """Prompts the user to enter an email address, and validates with a very long regular expression"""
+
+        # ----SETUP----
+        # ask for the email address as a freeform question instead of doing it ourselves
+        answer = await self.ask_freeform_question(
+            recipient, question_text, require_first_device
```
Comment (technillogue) [inline forest/core.py:1526]: this shouldn't ever require_first_device; remove it as an argument

**Review by technillogue — COMMENTED** [review]
looks great, congrats!

**Comment by technillogue** [comment]
you also need to fix mypy errors, lmk if you need help with that

**Comment by taygetea** [comment]
should all be fixed.

**Review by technillogue — APPROVED** [review]

**Pull request merged**